packages/client/hmi-client/src/components/responsive-matrix/pixi-utils.ts

```diff
@@ -0,0 +1,70 @@
+import { Resource, Texture, BaseTexture, SCALE_MODES, MIPMAP_MODES } from 'pixi.js';
+
+// taken from https://github.com/pixijs/pixijs/issues/6436#issuecomment-591695313
+export class CustomBufferResource extends Resource {
+	constructor(source, options) {
+		const { width, height, internalFormat, format, type } = options || {};
+
+		if (!width || !height || !internalFormat || !format || !type) {
+			throw new Error(
+				'CustomBufferResource width, height, internalFormat, format, or type invalid'
+			);
+		}
+
+		super(width, height);
+
+		(this as any).data = source;
+		(this as any).internalFormat = internalFormat;
+		(this as any).format = format;
+		(this as any).type = type;
```

Comment:
If you are adding/extending class attributes to `Resource`, maybe we can just desclare these class variables so you don't need to do `this as any` all over the place?

e.g.

```
export class X extends Resource {
  internalFormat: string
  type: string 
  ...

  constructor() { ... }
}
```

Reply:
sounds good will do